Sources/Bot/Network/Request.swift

```diff
@@ -0,0 +1,6 @@
+import Foundation
+
+public protocol RequestProtocol {
```

Comment:
I'd just use an extension on `Request`, not likely it will be anything else. And I'm guessing it should be possible to create a request for tests. If not - why not to go with a struct rather than protocol?

Reply:
The main purpose of this is to avoid having `Bot` module depending of `Vapor` at all, that would gives enough flexibility to change it to something else if needed.

Reply:
Interesting. I was thinking about this dependency for Stevenson as well, because it does not really _have_ to use Vapor, but decided that it's fine as the app is built with Vapor anyway and the stuff that Vapor already provides even just Futures makes it not to depend on anything else, like RAS. And considering Combine I wouldn't be surprised if in future Vapor encomorates it somehow as well on top of SwiftNIO futures (which feel a bit of clumsy to use sometimes because of event loops architecture).
Let's see how it goes 👍

Reply:
yeah tbh trying not to depend on Vapor but then depend on RAS instead kinda defeat the initial idea, doesn't it?

Right now your goal is if we decide to use something other than Vapor we can easily switch... but if we decide to use something other than RAS, e.g. Combine, we'd still need to switch too... and having to balance with both the `Future` and `SignalProducer` concepts back and forth makes the codebase more confusing imho.
Also my fear is that there's also a risk of adding bugs if there is some convoluted race condition between Vapor's EventLoop and RAS' scheduler that we didn't think about...

Personally I don't think bringing a whole dependency just for that, and adding a separation layer here that we won't likely need anyway (YAGNI) feels unnecessary, harder to debug and to reason about...

Reply:
@AliSoftware I agree with you but there's a limitation here, the current implementation of the bot uses RAS deeply and I'd like to have it here first up and running before making any significant refactor, that's why I'm bringing RAS but it's very likely that it will go away in the short-term. Not depending of Vapor was super easy to achieve that's why it's being done already, RAS will be done later on if it makes sense.

Reply:
Ok. In that case let's ship it that way then.

Still confused by your "the current implementation of the bot uses RAS deeply" since `ReactiveSwift` was added in the `Package.swift` in this very PR, so how could the implementation _before that PR_ depend on RAS then? (Or did you mean "just the current implementation of GitHubService, that I just added in this PR"?)

Reply:
Sorry, my plan is to move our current implementation (which is private ATM) here and start working from there instead of starting from scratch.